packages/bee-q/src/components/dropdown/bq-dropdown.tsx

```diff
@@ -0,0 +1,164 @@
+import { h, Component, Element, Event, EventEmitter, Prop, Listen } from '@stencil/core';
+
+import { FloatingUIPlacement } from '../../services/interfaces';
+
+/**
+ * @part base - The component's internal wrapper.
+ * @part trigger - The `div` element used to display the trigger element
+ * @part panel - The `div` element used to display the panel element (bq-panel)
+ */
+@Component({
+  tag: 'bq-dropdown',
+  styleUrl: './scss/bq-dropdown.scss',
+  shadow: true,
+})
+export class BqDropdown {
+  // Own Properties
+  // ====================
+
+  private trigger: HTMLElement;
+  private panelElement: HTMLBqPanelElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqDropdownElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** Distance (px) between the panel and the trigger element. */
+  @Prop({ reflect: true }) panelDistance?: number = 4;
+
+  /** Position of the panel */
+  @Prop({ reflect: true }) panelPlacement?: FloatingUIPlacement = 'bottom-start';
+
+  /**
+   * If true, panel is visible.
+   * You can toggle this attribute to show/hide the panel.
+   */
+  @Prop({ reflect: true }) panelOpen?: boolean = false;
+
+  /**
+   * Determines whether the scrollbar is visible or hidden within the panel.
+   */
+  @Prop({ reflect: true }) panelScrollbar?: boolean = false;
+
+  // Prop lifecycle events
+  // =======================
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /**
+   * Handler to be called to check if the `bq-panel` switches state (visible/hidden).
+   * @returns CustomEvent - with value `{ opened: boolean }`
+   */
+  @Event() bqPanelChange: EventEmitter<{ opened: boolean }>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentDidLoad() {
+    this.initFloatingUIFromPanel();
+  }
+
+  // Listeners
+  // ==============
+
+  /** On click outside the panel */
+  @Listen('click', { target: 'document', passive: true })
+  onClickOutsidePanel(event: MouseEvent) {
+    // close the panel on click, except click within the panel or within the trigger element
+    if (!event.composedPath().includes(this.panelElement) && !event.composedPath().includes(this.trigger)) {
+      this.panelElement.open = false;
+    }
+  }
+
+  /**
+   * If the user press on Escape button
+   * or `<bq-dropdown>` loses focus
+   * then hide the panel
+   */
+  @Listen('keyup', { target: 'window', passive: true })
+  onEscape(event: KeyboardEvent) {
+    if (event.key === 'Escape' || (event.key === 'Tab' && !event.composedPath().includes(this.el))) {
+      this.panelElement.open = false;
+    }
+  }
```

Comment:
We should skip if the panel is already closed.

```suggestion
  /**
   * Listens for the 'keyup' event on the window object
   * and closes the dropdown panel if the 'Escape' key or 'Tab' key outside the component is pressed.
   */
  @Listen('keyup', { target: 'window', passive: true })
  onEscape(event: KeyboardEvent) {
    if (!this.open) return;
    if (event.key === 'Escape' || (event.key === 'Tab' && !event.composedPath().includes(this.el))) {
      this.open = false;
    }
  }
```